tcplay: init at 2.0

kirushik opened the pull request:
###### Motivation for this change
###### Things done
- [x] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
  - [x] NixOS
  - [ ] OS X
  - [ ] Linux
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

kirushik committed:
tcplay: init at 2.0

groxxda commented on pkgs/tools/security/tcplay/default.nix:
```diff
@@ -0,0 +1,26 @@
+stdenv.mkDerivation rec {
+  name = "tcplay-2.0";
+
+  src = fetchurl {
```
Maybe switch to fetchFromGitHub

kirushik replied:
Oh, if I understand this correctly (and this is my first nix package, so I might as well not be) — `fetchFromGithub` checkouts the specified revision|branch from git directly.
Is it any better than downloading the official release tarball, as tcplay authors recommend to do [here](https://github.com/bwalex/tc-play#download-for-packaging)?

groxxda replied:
`fetchgit` does what you described.
`fetchFromGitHub` is just a wrapper around fetchzip and downloads the archive in tgz just like you manually do
It's declared [here](https://github.com/NixOS/nixpkgs/blob/master/pkgs/top-level/all-packages.nix#L200)

kirushik commented:
@groxxda Thanks for your clarifications, I've updated the `default.nix` to use `fetchFromGitHub`, rebased to the latest master and force-pushed (to keep the history intact).

Please have another look (and maybe double-check if it builds, if possible — as I won't have access to my NixOS machine till tomorrow unfortunately...)

groxxda commented:
This will not work since you also have to change the import from `fetchurl` to `fetchFromGitHub` :)

kirushik commented:
@groxxda Damn, my bad. That's what you're getting when writing code "from the top of your head", without actually testing it.

Is everything fine now from your standpoint, or do you want me to close it for now and reopen when I'll be able to make sure that everything is totally fine?

joachifm commented on pkgs/tools/security/tcplay/default.nix:
```diff
@@ -0,0 +1,27 @@
+{ stdenv, fetchFromGithub, cmake, pkgconfig, libgcrypt, libuuid, devicemapper, libudev, openssl, libgpgerror }:
+
+stdenv.mkDerivation rec {
+  name = "tcplay-2.0";
```
To make updating easier, you could factor out a `version` attribute and use it here and in `rev` below.

joachifm commented on pkgs/tools/security/tcplay/default.nix:
```diff
@@ -0,0 +1,27 @@
+
+  meta = {
+    description = "Free and simple TrueCrypt Implementation based on dm-crypt";
+    license = stdenv.lib.licenses.bsd2;
```
Please add `meta.platforms` and also consider maintaining the package --- adding new orphans makes no sense, really.

joachifm commented on pkgs/tools/security/tcplay/default.nix:
```diff
@@ -0,0 +1,27 @@
+
+  buildInputs = [ cmake pkgconfig libgcrypt libuuid devicemapper libudev openssl libgpgerror ];
+
+  configurePhase = ''
```
cmake's setup hook also defines a configure phase, are you sure it is necessary to override it?

joachifm commented on pkgs/tools/security/tcplay/default.nix:
```diff
@@ -0,0 +1,27 @@
+{ stdenv, fetchFromGithub, cmake, pkgconfig, libgcrypt, libuuid, devicemapper, libudev, openssl, libgpgerror }:
```
Should be `fetchFromGitHub` (note the capital H).

joachifm replied:
Minor style point: the preferred formatting of multi-line parameter specs is
```
{ stdenv, ...
, foo
, ...
}:
```

groxxda commented:
It's actually called `fetchFromGitHub` (capital H, sorry for missing that before) and the hash needs an update. 
Builds for me with these minor [fixes](https://github.com/groxxda/nixpkgs/commit/86b8d91945bf64d8c6e635813a69b22371b896e8.patch)

kirushik commented:
Guys, thank you for your feedback — but it looks like my affair with NixOS hadn't worked out, and I'm back on OpenSUSE.
This probably makes me not the best candidate to take care of maintaining this package.

I'm sorry for the inconvenience; please close this PR if needed.

bjornfor commented:
@kirushik: May I ask what made you abandon NixOS? Any issues in particular?

kirushik commented:
@bjornfor Yes, sure.
That's mostly my "special preferences" in setting up a working environment.
I'm a heavy `i3wm` addict, and I like to lightweight GUI tools, mostly from from XFCE (thunar, squeeze, ristretto) and Enlightment (terminology).

For such a setup I've completely failed to, say, make icons in Thunar work, when launched from i3. When I start XFCE DE, they are there, but if I start an i3 session — they're not, and I failed to figure this out (yes, I have tried to link icon folders in my configuration, but I should've made a mistake somewhere...)
`squeeze` archive manager (http://squeeze.xfce.org/) is not packaged for Nix, and given my troubles with icons, I didn't even want to think about all the funny issues which will pop up when I'd want to integrate squeeze operations in Thunar without full XFCE session...

That's where I've almost exhausted my "3-4 afternoons" time budget for installing a new OS on my laptop, and decided to go with something I understand better.
I cannot stress enough, that all those are (most probably) issues with my understanding of Nix, not the OS itself — and I will certainly give it another try in a year or so.

bjornfor commented:
@kirushik: Thanks for the info! I hope next time you try NixOS that it has improved enough so you will make the jump :-)
